### DISCUSSION TITLE
Danish layout

### DISCUSSION THREAD

**Discussion by eyJhb** [8VIM/8VIM#96] [Language Layout]
There is a need for æ ø and å for a danish layout.
I am not sure of the best way to place this. Maybe there could be a combo to translate aa to å, oe to ø and se to æ?

Here is a frequency of chars for danish layout - https://www.sttmedia.com/characterfrequency-danish

**Comment by WuerfelDev** [comment]
For reference the danish layout from 8pen from 2011

![Screenshot_20201114-233328__01](https://user-images.githubusercontent.com/12686819/99158358-2cb55100-26d2-11eb-8605-143b81fb6ca2.png)

![Screenshot_20201114-233403__01](https://user-images.githubusercontent.com/12686819/99158359-2d4de780-26d2-11eb-8035-3ab3680000ab.png)

**Comment by Glitchy-Tozier** [comment]
Sounds like #65 should be settled fist, as `æ`, `ø`, and `å` maybe could be inserted the way diacritics would.

**Comment by Glitchy-Tozier** [comment]
You might be interested in this:
https://github.com/flide/8VIM/discussions/138

**Comment by adaschma** [comment]
Hi @eyJhb 
I've configured the Danish diacritics into the standard 8pen layout. What is still needed is the UI to switch different layouts in 8vim. If anybody would like to use it before that is done, I can offer a version of 8vim with the Danish diacritics just like I do for the German layout.

**Comment by flide** [comment]
Hi All,

With the help of @adamschmalhofer 's contribution, I have been able to setup a framework for adding layouts easily.

To add a new layout following needs to be done:

    Develop a keyboard-data xml file.
    Name the file with following format [{language-iso-code}_{font-code}_{layout-name}]
    Place the file in the raw folder

    Language code is the 2 character code for the language as per ISO standard
    Font code is one of the following strings, and as of now only "regular" is going to function properly : "regular", "bold", "italic", "underline"
    Layout name is the display name of the layout

In the settings menu the layout would be presented as "Layout name (Language)"

The characters for the layout are now part of the keyboard-data file itself.
Please check the "he_regular_hebrew" file in the raw folder for reference.

I will request all to open one PR per layout.
